lib/kubernetes-deploy/renderer.rb

```diff
@@ -60,12 +60,14 @@ def render_partial(partial, locals)
       # Note that JSON is a subset of YAML.
       JSON.generate(docs.children.first.to_ruby)
     rescue PartialNotFound => err
-      raise InvalidPartialError, err.message
+      # get the filename from the first parent, not the missing partial itself
+      raise err if err.filename == partial
+      raise InvalidPartialError.new(err.message, filename: partial, content: expanded_template || template)
```

Comment:
Shouldn't the underlying code raise the correct error type? Instead of rescuing one type and raising either one or the other?

Reply:
This is transforming the data, not just the error type. `rescue PartialNotFound` needs to be distinct because it is the _first parent_ (potentially of a long chain) that has the relevant filename and template content. So we hit this case in the missing child (in which case we skip it, L64) and the _first_ parent (in which case we grab the filename and content, L65), and then after that we just add the additional parent name to the list (L67), like we would for any other type of error.